x-pack/plugins/ml/public/application/timeseriesexplorer/timeseriesexplorer.js

```diff
@@ -844,10 +847,22 @@ export class TimeSeriesExplorer extends React.Component {
 
   componentDidUpdate(previousProps) {
     if (previousProps === undefined || previousProps.selectedJobId !== this.props.selectedJobId) {
+      const selectedJob = mlJobService.getJob(this.props.selectedJobId);
       this.contextChartSelectedInitCallDone = false;
-      this.setState({ fullRefresh: false, loading: true }, () => {
-        this.loadForJobId(this.props.selectedJobId);
-      });
+      getSourceIndicesWithGeoFields([selectedJob], this.props.dataViewsService)
```

Comment:
There are many other props in the component that can be updated. Before calling `getSourceIndicesWithGeoFields`, you should first check that `selectedJob` has been updated

Reply:
The if statement above checks if the selected job id has been updated `previousProps.selectedJobId !== this.props.selectedJobId` - wouldn't that cover that scenario?